impeller/entity/contents/atlas_contents.cc

```diff
@@ -100,7 +100,9 @@ bool AtlasContents::Render(const ContentContext& renderer,
   for (size_t i = 0; i < texture_coords_.size(); i++) {
     auto sample_rect = texture_coords_[i];
     auto matrix = transforms_[i];
-    auto color = (colors_.size() > 0 ? colors_[i] : Color::Black());
+    auto color = (colors_.size() > 0
```

Comment:
I don't believe that this is correct - for drawAtlas we blend the colors with the texture sample which must be done in the shader: https://api.flutter.dev/flutter/dart-ui/Canvas/drawAtlas.html

> Blend each rectangular portion of the image specified by an entry in the rects argument with its associated entry in the colors list using the blendMode argument (if a color is specified). In this part of the operation, the image part will be considered the source of the operation and the associated color will be considered the destination.